content/tutorials/advanced/code/AdvancedTutorialSampleJava.java

```diff
@@ -17,134 +17,328 @@
 }
-//#setup-users
+//#login-endpoint
 
-//#setup-users-and-admins
-{
-  setUp(
-    users.injectOpen(rampUsers(10).during(10)),
-    admins.injectOpen(rampUsers(2).during(10))
-  ).protocols(httpProtocol);
+//#with-authentication-headers-wrapper
+// Add authentication header if an access token exists in the session
+// Reference:
+// https://docs.gatling.io/reference/script/protocols/http/request/#headers
+public static final HttpProtocolBuilder withAuthenticationHeader(
+    HttpProtocolBuilder protocolBuilder) {
+  return protocolBuilder.header(
+      "Authorization",
+      session -> Optional.ofNullable(session.getString("AccessToken")).orElse(""));
 }
-//#setup-users-and-admins
-  }
+//#with-authentication-headers-wrapper
 
-//#feeder
-FeederBuilder.Batchable<String> feeder =
-  csv("search.csv").random(); // 1, 2
-
-ChainBuilder search = exec(http("Home")
-  .get("/"))
-  .pause(1)
-  .feed(feeder) // 3
-  .exec(http("Search")
-    .get("/computers?f=#{searchCriterion}") // 4
-    .check(
-      css("a:contains('#{searchComputerName}')", "href")
-        .saveAs("computerUrl") // 5
-    )
-  )
-  .pause(1)
-  .exec(http("Select")
-    .get("#{computerUrl}")) // 6
-  .pause(1);
-//#feeder
-
-  public static final class BrowseLoopSimple {
-
-//#loop-simple
-private static ChainBuilder gotoPage(int page) {
-  return exec(http("Page " + page)
-    .get("/computers?p=" + page))
-    .pause(1);
+//#homepage-endpoint
+public class WebEndpoints {
+  // Define the home page request with response status validation
+  // Reference: https://docs.gatling.io/reference/script/protocols/http/request/#checks
+  public static final HttpRequestActionBuilder homePage =
+      http("HomePage")
+          .get("https://ecomm.gatling.io")
+          .check(status().in(200, 304)); // Accept both OK (200) and Not Modified (304) statuses
 }
+//#homepage-endpoint
+
+  public static class ScenarioGroupsWrapper {
 
-ChainBuilder browse =
-  exec(
-    gotoPage(0),
-    gotoPage(1),
-    gotoPage(2),
-    gotoPage(3),
-    gotoPage(4)
-  );
-//#loop-simple
+    private static final int minPauseSec =
+      Integer.getInteger("minPauseSec", 5); // Minimum pause between actions
+    private static final int maxPauseSec =
+      Integer.getInteger("maxPauseSec", 15); // Maximum pause between actions
+    
+    public static final HttpRequestActionBuilder login =
+      http("Login")
+          .post("/login");
+
+    public static final HttpRequestActionBuilder loginPage =
+      http("LoginPage").get(pageUrl + "/login").check(status().in(200, 304));
+
+//#authenticate-group
+public class ScenarioGroups{
+  private static final FeederBuilder<Object> usersFeeder =
+    jsonFile("data/users_dev.json").circular();
+    // Define authentication process
+  public static final ChainBuilder authenticate =
+    group("authenticate")
+      .on(loginPage, feed(usersFeeder), pause(5, 15), login);
   }
+//#authenticate-group
```

Comment:
- Remove extra indent
- Hide parent class
```suggestion
public class ScenarioGroups{
//#authenticate-group
private static final FeederBuilder<Object> usersFeeder = jsonFile("data/users_dev.json").circular();
// Define authentication process
// Reference: https://docs.gatling.io/reference/script/core/group/
public static final ChainBuilder authenticate =
  group("authenticate")
    .on(loginPage, feed(usersFeeder), pause(5, 15), login);
//#authenticate-group
  }
```